### ISSUE TITLE
Request: add the whois and gettext packages

### ISSUE THREAD

**Issue by viktorradnai** [stevearc/pypicloud-docker#31]
I would like to use Pypicloud with Kubernetes, and for this reason I would like to pass in some values from secrets. These will become environment variables, and the `envsubst` command (from the `gettext` package) can be used to template them into `config.ini`. You can see my implementation at https://github.com/HummingbirdTechGroup/helm-pypicloud. I am trying to avoid using a different image for the init-container if possible.

It may also make sense to add the `whois` package which provides the `mkpasswd` command that can be used to generate SHA512 password hashes for config.ini. Although I have opened a separate PR for pypicloud for `pypicloud-gen-password` to support reading password values from stdin, the mkpasswd command is written in C and works much faster.

**Comment by stevearc** [comment]
If you're trying to pass configuration in to pypicloud using environment variables, you may be interested in this change that I made for someone else: https://github.com/stevearc/pypicloud/commit/f9ca8312fdbf3c794030395646e45991b4f4e25a
This was a one-off request to allow passing a single setting in, but this could easily be expanded to other values. If you think this fits your needs better, feel free to open another PR adding this to more locations.

I like the approach in your PR to make the `gen-password` command read from stdin. I'd prefer to stick to that and avoid adding other packages to the containers unless it's absolutely necessary.

**Comment by viktorradnai** [comment]
> If you're trying to pass configuration in to pypicloud using environment variables, you may be interested in this change that I made for someone else: [stevearc/pypicloud@f9ca831](https://github.com/stevearc/pypicloud/commit/f9ca8312fdbf3c794030395646e45991b4f4e25a)
> This was a one-off request to allow passing a single setting in, but this could easily be expanded to other values. If you think this fits your needs better, feel free to open another PR adding this to more locations.

Unfortunately this does not work if I use an init container to hash the password, as changes to the environment variables inside the init container are not inherited by other containers in the pod. It could only be done by adding the password hashing code to `pypicloud-uwsgi.sh` and exporting the hashed values as `PPC_USER_xxx_PASSWORD` environment variables there. Personally I consider this messy.
 
> I like the approach in your PR to make the `gen-password` command read from stdin. I'd prefer to stick to that and avoid adding other packages to the containers unless it's absolutely necessary.

My preferred approach would have been to use Helm to generate the hash, but the only relevant function for that in the Helm templating language is `htpasswd` -- it can be used to create a `bcrypt` hash with a dummy username (See http://masterminds.github.io/sprig/crypto.html). This did not work out of the box because it would have required installing the `bcrypt` library in your image. 

Another way would have been to add a feature request to Helm for to implement sha512crypt but it would have taken too long for me. So I went with the init container and initially used an Ubuntu 20.04 base image with these extra packages installed. However I'd really like to avoid having to maintain another Docker image for the init container just to be able to hash these passwords, which is why I requested these packages to be added. 

Since you've merged my other [PR](https://github.com/stevearc/pypicloud/pull/253) (thanks :) ) only `gettext` package is required to provide the `envsubst` command that does the environment substitution - replacing `$ENV_VAR` with the value of `$ENV_VAR` in `config.ini`. I think this could be useful for other cases such as Docker Compose.

To sum it up, we have the following options to make password hashing work in my use case (in my order of preference):

1. You add the `bcrypt` library and I hash the passwords using Helm
2. You add the `gettext` package and I hash the passwords using an init container based on your image (using `pypicloud-gen-password` and `envsubst`)
3. You add the password hashing code and `envsubst` (or some other mechanism eg the `PPC_*` env vars) to the `pypicloud-uwsgi.sh` script and the init container is not required
4. I create and maintain another docker image for the init container (possibly based on alpine, busybox or some other lightweight image) 
 
Please let me know your thoughts / preferences on the above.

**Comment by stevearc** [comment]
After some tests, it looks like adding the bcrypt library bloats the images the least
alpine: 247MB -> 248MB
baseimage: 722MB -> 722MB

Compared to gettext
alpine: 247MB -> 252MB
baseimage: 722MB -> 761MB

I'll add bcrypt and you can let me know if that's enough to get it working for you

**Comment by viktorradnai** [comment]
Thanks for this @stevearc and sorry for the delay in testing it. Apart from the commit above, you also need to `pip3 install bcrypt` as the deb / apk package does not contain the Python library. I have made the necessary adjustments to my Helm chart so once you add the missing library and cut the next release it will work.

**Comment by stevearc** [comment]
Added!

**Comment by viktorradnai** [comment]
Thanks, this now works using `stevearc/pypicloud:latest`. Feel free to close this ticket or leave it open until it makes it into a tagged release.

**Comment by viktorradnai** [comment]
Closing issue as new tagged versions were released with this.
